static/pane.ts

```diff
@@ -0,0 +1,195 @@
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import _ from 'underscore';
+import { Container } from 'golden-layout';
+import * as monaco from 'monaco-editor';
+
+import { FontScale } from './fontscale'
+import { BasePaneState, PaneCompilerState } from './pane.interfaces';
+import { SiteSettings } from './settings.interfaces';
+
+/**
+ * Basic container for a tool pane in Compiler Explorer
+ *
+ * Type parameter E indicates which monaco editor kind this pane hosts. Common
+ * values are monaco.editor.IDiffEditor and monaco.ICodeEditor
+ */
+export abstract class Pane<E extends monaco.editor.IEditor> {
+    compilerInfo: PaneCompilerState;
+    container: Container;
+    domRoot: JQuery;
+    topBar: JQuery;
+    eventHub: any /* typeof hub.createEventHub() */;
+    selection: monaco.Selection;
+    editor: E;
+    fontScale: typeof FontScale
+    isAwaitingInitialResults: boolean = false;
+    settings: SiteSettings | {} = {};
+
+    /**
+     * Base constructor for any pane. Performs common initialization tasks such
+     * as registering standard event listeners and lifecycle handlers.
+     *
+     * Overridable for implementors
+     */
+    protected constructor(hub: any /* Hub */, container: Container, state: BasePaneState) {
+        this.container = container;
+        this.eventHub = hub.createEventHub();
+        this.domRoot = container.getElement();
+        this.initializeDOMRoot();
+
+        const editorRoot = this.domRoot.find('.monaco-placeholder')[0];
+        this.createEditor(editorRoot);
+
+        this.selection = state.selection
+        this.compilerInfo = {
+            compilerId: state.id,
+            compilerName: state.compilerName,
+            editorId: state.editorid,
+        }
+        this.fontScale = new FontScale(this.domRoot, state, this.editor);
+        this.topBar = this.domRoot.find('.top-bar');
+
+        this.registerButtons(state);
+        this.registerStandardCallbacks();
+        this.registerCallbacks();
+        this.setTitle();
```

Comment:
Missing a call to `this.registerOpeningAnalyticsEvent()`?

Reply:
Yep, thanks!